sleap_roots/points.py

```diff
@@ -190,14 +219,26 @@ def get_root_vectors(start_nodes: np.ndarray, end_nodes: np.ndarray) -> np.ndarr
         An array of vectors with shape (instances, 2), representing the vector from start
         to end for each instance.
     """
+    # If start or end nodes is not a 2D array, make it one
+    start_nodes = np.atleast_2d(start_nodes)
+    end_nodes = np.atleast_2d(end_nodes)
+
+    # Handle too many dimensions
+    if start_nodes.ndim > 2 or end_nodes.ndim > 2:
+        raise ValueError(
+            "start_nodes and end_nodes should have maximum dimension of 2."
+        )
+
     # Ensure that the start and end nodes have the same shapes
-    if start_nodes.shape != end_nodes.shape:
-        raise ValueError("start_nodes and end_nodes should have the same shape.")
-    # Handle single instances with shape (2,)
-    if start_nodes.ndim == 1:
-        start_nodes = start_nodes[np.newaxis, :]
-    if end_nodes.ndim == 1:
-        end_nodes = end_nodes[np.newaxis, :]
+    if start_nodes.shape[0] != end_nodes.shape[0]:
```

Comment:
why not check the whole shape instead of just the first dimension?

```
if start_nodes.shape != end_nodes.shape:
    raise ValueError("start_nodes and end_nodes must have the same shape.")

if start_nodes.shape[1] != 2:
    raise ValueError("Each node must have exactly 2 coordinates (x, y).")
```